docs/calyx-py.md

```diff
@@ -1,17 +1,120 @@
+
+```python
+{{#include ../calyx-py/test/builder_example.py:group_def}}
 ```
 
+Now, we'll initialize our registers. You can access cell ports using dot notation. Notably, port names that are also reserved keywords in Python such as `in` are followed by an underscore.
+
+```python
+{{#include ../calyx-py/test/builder_example.py:assigns}}
+```
+
+As mentioned in the comments above, the Calyx builder will try to infer the bitwidth of constants. In case this doesn't work and you run into problems with this, you can provide the constant's size like so:
+
+```python
+{{#include ../calyx-py/test/builder_example.py:const}}
+```
+
+Calyx groups use a latency-insensitive go/done interface. Writing a group's done condition with the builder is pretty similar to doing so in Calyx, except that the `?` used for guarded assignments is now `@` (due to conflicting usage of `?` in Python).
+
+```python
+{{#include ../calyx-py/test/builder_example.py:done}}
+```
+
+In order to use the ports of cells in our `main` component within the code for our component, we'll expose the adder's output port by explicitly constructing it using the `calyx-py` AST.
+
+```python
+{{#include ../calyx-py/test/builder_example.py:bare}}
+```
+
+Now, when we want to use the output port of our adder, we can do so easily:
+
+```python
+{{#include ../calyx-py/test/builder_example.py:bare_use}}
+```
+
+In order to add continuous assignments to your program, use the construct `with {component}.continuous:`.
+
+```python
+{{#include ../calyx-py/test/builder_example.py:continuous}}
+```
+
+To access a component's ports while defining it, like we did above, we use the method `this()`.
+
+```python
+{{#include ../calyx-py/test/builder_example.py:this}}
+```
+
+Lastly, we'll construct the control portion of this Calyx program. It's pretty simple; we're sequencing two groups. Sequences of groups are just Python lists:
+
+```python
+{{#include ../calyx-py/test/builder_example.py:return}}
+```
+You can also use the builder for generating parallel control blocks. Just encapuslate the parallel groups (say, `A`, `B`, and `C`) with the `par` keyword. For instance, the above code with some parallel groups in it might look like
+
+```python
+    main.control += [
+        update_operands,
+        compute_sum,
+        par(A, B)
+    ]
+```
+
+After making our modifications to the `main` component, we'll return the program.
+
+```python
+{{#include ../calyx-py/test/builder_example.py:return}}
+```
+
+Finally, we can emit the program we built.
+
+```python
+{{#include ../calyx-py/test/builder_example.py:emit}}
+```
+
+In this program.`build()` serves both as a way to define the `main` component as well as organize the Calyx program we generated. Since this program was so simple, we didn't need to factor out our component definitions. For Calyx programs with multiple components or more complex operations, factoring definitions or procedures into their own functions is useful.
```

Comment:
Not sure what this little bit does for us, haha

Reply:
I wanted to cover multi-component designs (even if it's cursory) so that folks have a little more to go off of if they wanted to use the builder for that. Do you think the last paragraph should be removed?

Reply:
If you want to cover multi-component designs (which I think is a good idea btw), I think the code needs a multi-component design. This is neither here nor there; it sorta just reads like, "You can split this into multiple Python `def`s. We haven't, but you could." A truism, IMO